### PR TITLE
Simplifying the verification rule

### PR DESCRIPTION
**Pull request by FauziAkram**
Simplifying the verification rule by removing two operations.

Passed STC non-reg:
LLR: 2.94 (-2.94,2.94) <-1.75,0.25>
Total: 46368 W: 12124 L: 11914 D: 22330
Ptnml(0-2): 198, 5170, 12210, 5436, 170
https://tests.stockfishchess.org/tests/view/65bf71fcc865510db027e1ff

Passed LTC non-reg:
LLR: 2.94 (-2.94,2.94) <-1.75,0.25>
Total: 214764 W: 53946 L: 53924 D: 106894
Ptnml(0-2): 116, 23288, 60560, 23294, 124
https://tests.stockfishchess.org/tests/view/65bfbcabc865510db027e8f9

Here are the results of the zugzwang test suite, filled by personally observing each position until the patch was able to find the correct evaluation (fail high in most of the positions, and draw in a single position (pos.20).
Engines were set to use only one core, and a maximum time of 120 sec.

image

So in conclusion I didn't observe any regression in detecting zugzwang positions, the only position that took a little longer was pos. 17, but since in other positions the new patch sometimes took less time to find the solution, I don't think it's relevant.
Also, for the positions in which the patches took longer than 2 seconds to solve, I ran them twice, and got always very similar results.

Test Suite used: https://github.com/official-stockfish/Stockfish/files/1558563/Zugzwang-Allgeuer.zip

bench: 1478189

### CODE REVIEW

**Comment by FauziAkram** [comment]
> Can you please add here a link to the zugzwang test suite you used, thanks.

Done. Added in the PR draft itself.
Anyway here is the link: https://github.com/official-stockfish/Stockfish/files/1558563/Zugzwang-Allgeuer.zip

**Comment by Disservin** [comment]
I honestly don't really see any point in changing this tbh. This verification search was introduced with non regression bounds and it was never a real gainer in fishtest.. so you can reduce this as much as you want and you'd still be getting patches which pass non regression...

**Comment by FauziAkram** [comment]
Yes, I know that, that's why I ran the additional zugzwang suite test.
This should help to prove that the zugzwang detection, is still effective even with this simpler version.

Anyway, I don't have any strong opinion about this patch, if you think it's better to decline it, I am ok with that.

**Comment by mstembera** [comment]
Perhaps we should add some comment so it won't get simplified in the future?

**Comment by FauziAkram** [comment]
> Perhaps we should add some comment so it won't get simplified in the future?

What if in the future, SF reaches a state where this verification is not needed anymore, and even by removing it completely, SF has no regression related to zugzuwang positions?
In that case, what would be the benefit of keeping it?

**Comment by mstembera** [comment]
I'm not suggesting the comment says don't simplify under any circumstances.   It should just explain what's not obvious about this code.

**Comment by FauziAkram** [comment]
> I'm not suggesting the comment says don't simplify under any circumstances. It should just explain what's not obvious about this code.

Then I completely agree, do you mind suggesting a comment?
I can add it to my pending list of cleanups/small improvements, and once I have enough of them, I will open a PR.

**Comment by mstembera** [comment]
I would prefer someone who actually understands the details like you or @Disservin write it.  I only know that things like https://github.com/official-stockfish/Stockfish/pull/5038#issuecomment-1936240683 and https://github.com/official-stockfish/Stockfish/pull/5038#issuecomment-1936268035 are not obvious when just reading the code.

**Comment by pb00067** [comment]
I wonder why position 24 (6Q1/8/8/7k/8/8/3p1pp1/3Kbrrb w - - 0 1 ) failed on your tests,
on my computer it get resolved in less than 1 second as well in master as in the patch.

**Comment by FauziAkram** [comment]
> I wonder why position 24 (6Q1/8/8/7k/8/8/3p1pp1/3Kbrrb w - - 0 1 ) failed on your tests, on my computer it get resolved in less than 1 second as well in master as in the patch.

1) My whole test was with a single thread
2) If the engine suggests Qg7 but shows an eval of 0.00 I don't consider it to be solved until it gets out of the 0.00

**Comment by pb00067** [comment]
I use one single thread too and it resolves to mate within 1 second (current master):
 
```
2024-02-12 11:21:23.730-->1:setoption name Hash value 1024
2024-02-12 11:21:23.736*1*Child Process Prio Adj:  PID 13216 conhost.exe
2024-02-12 11:21:23.737-->1:isready
2024-02-12 11:21:23.881<--1:readyok
2024-02-12 11:21:23.944**Time for starting engines for analysis:406 ms
2024-02-12 11:21:23.945-->1:isready
2024-02-12 11:21:23.945<--1:readyok
2024-02-12 11:21:24.159<--1:info string NNUE evaluation using nn-baff1edbea57.nnue
...
2024-02-12 11:21:24.334<--1:info depth 56 seldepth 11 multipv 1 score cp 0 nodes 403091 nps 3387319 hashfull 0 tbhits 0 time 119 pv g8e8 h5h6 e8g8 h6h5 g8e6 h5g5 e6f7 g5g4 f7f8 g4g5
2024-02-12 11:21:24.337<--1:info depth 57 seldepth 5 multipv 1 score cp 0 nodes 416675 nps 3415368 hashfull 0 tbhits 0 time 122 pv g8e8 h5h6 e8g8 h6h5
2024-02-12 11:21:24.340<--1:info depth 58 seldepth 6 multipv 1 score cp 0 nodes 456159 nps 3455750 hashfull 0 tbhits 0 time 132 pv g8e8 h5h6 e8g8 h6h5 g8e8
2024-02-12 11:21:24.343<--1:info depth 59 seldepth 8 multipv 1 score mate 4 nodes 621342 nps 3570931 hashfull 0 tbhits 0 time 174 pv g8g7 h5h4 g7g6 h4h3 g6g5 h3h2 g5h4
2024-02-12 11:21:24.346<--1:info depth 60 seldepth 8 multipv 1 score mate 4 nodes 657632 nps 3633325 hashfull 0 tbhits 0 time 181 pv g8g7 h5h4 g7g6 h4h3 g6g5 h3h2 g5h4
```
Maybe its because of the hashsize, which hashsize did you use?

**Comment by FauziAkram** [comment]
128Mb hash size

**Comment by FauziAkram** [comment]
Is this where I check the hash size I'm using (Arena GUI)?
image

**Comment by peregrineshahin** [comment]
Yes....

**Comment by vondele** [comment]
I'll close this one, I don't think we have a compelling reason to merge.
